src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -634,25 +634,25 @@ static int32_t SimpleStartsWith_Iterators(UCollationElements* pPatternIterator,
     {
         if (movePattern)
         {
-            patternElement = ucol_next(pPatternIterator, &errorCode);
+            patternElement = forwardSearch ? ucol_next(pPatternIterator, &errorCode) : ucol_previous(pPatternIterator, &errorCode);
```

Comment:
>forwardSearch [](start = 29, length = 13)

You may try to avoid checking every time  during the iteration by initializing something like the following

int32_t (*searchNextOfPrevious)(UCollationElements*, UErrorCode*) = forwardSearch ? ucol_next : ucol_previous;

And then you call searchNextOfPrevious directly.

Reply:
Do you mean from readability or performance perspective?

I've added the `inline` keyword so the method gets always inlined. It's invoked with two consts: `TRUE` and `FALSE` and the `C` compiler removes this check in the calling method body.

Reply:
I was talking about the performance. if you confirm the generated C compiler code is doing what you have said, then we should be fine.

Reply:
I did not look at the generated assembly code, but I've run the benchmarks and ensured that `StartsWith|IsPrefix` performance is the same.